design-toolkit/components/src/components/drawer/types.ts

```diff
@@ -0,0 +1,186 @@
+}>;
+
+export interface DrawerContainerProps
+  extends PropsWithChildren<{ className?: string }> {}
+
+export interface DrawerProviderProps extends PropsWithChildren {
+  onStateChange?: (drawerId: Key, state: DrawerState) => void;
+}
+
+export interface DrawerLayoutProps
+  extends DrawerContainerProps,
+    Partial<Record<DrawerPlacement, DrawerState>> {
+  /**
+   * Which drawers should extend to full container dimensions.
+   * Determines the overall layout structure and drawer relationships in regard to space.
+   *
+   * @default 'left right'
+   */
+  extend?: DrawerLayouts;
+  /**
+   * Determines how drawer interact with the main content area and overall layout:
+   *
+   * - `'push'`: Drawer pushes the main content aside, reducing its available width.
+   *   Content area shrinks to accommodate the panel space.
+   *   If no placements are defined for push, the default behavior for a drawer is to float over the main content without affecting its layout or dimensions.
+   *   Content remains at full width, panel appears as an overlay.
+   */
+  push?: DrawerPlacement;
+  classNames?: DrawerClassNames;
+}
+
+export interface DrawerProps extends DrawerContainerProps {
+  id: Key;
+  placement: DrawerPlacement;
+  size?: DrawerSize;
+  isOpen?: boolean;
+  defaultSelectedMenuItemId?: Key;
+  onOpenChange?: OnOpenChangeCallback;
+  onStateChange?: (state: DrawerState) => void;
+}
+
+export type OnOpenChangeCallback = ((isOpen: boolean) => void) | undefined;
+
+export interface DrawerMenuProps extends DrawerContainerProps {
+  position?: 'start' | 'middle' | 'end';
+}
+
+export interface DrawerTriggerProps extends DrawerContainerProps {
+  for: Key;
+  behavior?: 'open' | 'close' | 'toggle';
+  children: ReactElement<DOMAttributes<FocusableElement>, string>
+}
+
+export interface DrawerMenuItemProps {
+  className?: string;
+  children: ReactElement<DOMAttributes<FocusableElement>, string>;
+  id?: Key;
+}
+
+export interface DrawerPanelProps extends DrawerContainerProps {
+  id?: Key;
+}
+
+export interface DrawersContextValue {
+  drawerStates: Record<Key, DrawerState>;
+  toggleDrawer: (drawerId: Key) => void;
+  openDrawer: (drawerId: Key, menuItemId?: Key) => void;
+  closeDrawer: (drawerId: Key) => void;
+  getDrawerState: (drawerId: Key) => DrawerState;
+  registerDrawer: (
+    initialState: DrawerState,
+    callbacks?: {
+      onOpenChange?: OnOpenChangeCallback;
+      onStateChange?: (state: DrawerState) => void;
+    },
+  ) => void;
+  selectedMenuItemId?: Key;
+  selectMenuItem: (drawerId: Key, menuItemId?: Key) => void;
+  isSelectedMenuItem: (selectedMenuItemId?: Key, menuItemId?: Key) => boolean;
+}
+
+export interface DrawerContextValue {
+  state: DrawerState;
+}
+
+/**
+ * Extended Drawer Layout Configurations
+ *
+ * The layout system supports four different drawer extension modes that determine
+ * how drawers are arranged and which drawers extend to the full container dimensions.
+ *
+ * extend: "tall"
```

Comment:
This is a little confusing since, according to the actual type below, I think this should say "left right"?

```suggestion
 * extend: "left right"
```

Reply:
good call - I was experimenting with some ideas here